### PR TITLE
Update 04-builder.md

### PR DESCRIPTION
**Pull request by davidmfinol**
#### Changes

- Include an example build script to help users get started with their custom `buildMethod`.

#### Checklist

- [x] Read the contribution [guide](../CONTRIBUTING.md) and accept the [code](../CODE_OF_CONDUCT.md) of conduct
- [x] Readme (updated or not needed)
- [x] Tests (added, updated or not needed)

### CODE REVIEW

File: docs/github/v2/04-builder.md [hunk 1]
```diff
@@ -140,19 +140,27 @@ _**default:** `build`_
...
   with:
     buildMethod: EditorNamespace.BuilderClassName.StaticBuildMethod
 ```
 
+To use a modified version of the default Unity Builder build script, you can put [BuildScript.cs](https://gist.github.com/davidmfinol/bdd6bdb26dbd8d949db0993d0e070742) in your `Assets/Editor/UnityBuilderAction` directory and reference it:
+
+```yaml
+- uses: game-ci/unity-builder@v2
+  with:
+    buildMethod: UnityBuilderAction.BuildScript.Build
```
Comment (webbertakken) [inline docs/github/v2/04-builder.md:161]: Actually it wouldn't make sense to have a namespace in your codebase that's using a namespace that is so very specific for reason of not being ambiguous or being overridden.

My suggestion would be to stay with a single example and just explain that you can use our script as an example.

I also don't think it's useful to create a second source of truth for the build script honestly as it changes from time to time and the docs would become outdated fast and nobody would be able to update it in your gist.

Comment (davidmfinol) [reply docs/github/v2/04-builder.md:161]: The problem with our current build script is that it's not easy for users to copy, so if users want to use their own `buildMethod`, they either have to a) spend time re-implementing functionality that we already have in our build script b) copy over multiple files and directories, with all the extra complexity that entails.

You're right that having two sources of truth isn't ideal, so how about if I refactor our current build script down to a single file that could be easily dropped into any project?
It would end up looking very similar to the gist, but it would replace https://github.com/game-ci/unity-builder/tree/main/dist/default-build-script
I'd also spend some extra time testing to make sure none of the current functionality breaks.

Comment (webbertakken) [reply docs/github/v2/04-builder.md:161]: I agree that it's currently not easy to copy it.

Personally I'm expecting the script to cover a whole lot more than just versioning over time so I'm also not sure about turning it into one file (at least in the action).

Perhaps two sources of truth are fine, as the script you're providing is essentially a starting point for implementers. Not per se a should-work-for-all solution. Perhaps we can keep the code snippet part of the documentation as we have code highlighting in our docs and commenting functionality on GitHub.

What do you think?

Comment (davidmfinol) [reply docs/github/v2/04-builder.md:161]: You're suggesting having BuildScript.cs in the documentation repo?

I think that could work.

Comment (webbertakken) [reply docs/github/v2/04-builder.md:161]: Yea I think that works. Or we could even embed it in a markdown file in a code block, like other examples.

Either works for me

**Review by webbertakken — APPROVED** [review]
